### PR TITLE
[hacktoberfest] Model `license:other` + Dataset `languages`  challenges

### PR DESCRIPTION
**Pull request by Wauplin**
First discussed on [slack](https://huggingface.slack.com/archives/C01BWJU0YKW/p1695824770210009) (private link).

Adding some instructions + a table. Let's iterate from this :) 

See rendered file here: https://github.com/huggingface/hub-docs/blob/hacktoberfest-challenges-license-other/hacktoberfest_challenges/model_license_other.md

### CODE REVIEW

**Comment by davanstrien** [comment]
I made a few quick suggestions for style. Will take another look on Monday morning :)

File: hacktoberfest_challenges/model_license_other.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
+
```
Comment (osanseviero) [inline hacktoberfest_challenges/model_license_other.md:1]: Please put the license in the documentation images dataset on the hub

File: hacktoberfest_challenges/model_license_other.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+In this challenge, we will focus on models that as the license `"other"`. This means that the model has a custom license defined by the author. For example, LLama2 models have their own license [llama-2-community-license](https://ai.meta.com/llama/license). Historically, this information was not available in the model card metadata. This is now possible by adding the license name and a URL to it:
```
Comment (osanseviero) [inline hacktoberfest_challenges/model_license_other.md:8]:
```suggestion
In this challenge, we will focus on models that as the license `"other"`. This means that the model has a custom license defined by the author. For example, Llama 2 models have their own license [llama-2-community-license](https://ai.meta.com/llama/license). Historically, this information was not available in the model card metadata. This is now possible by adding the license name and a URL to it:
```

Comment (osanseviero) [inline hacktoberfest_challenges/model_license_other.md:8]: Llama 2 license is now supported among the official licenses, so maybe it makes sense to put an unsupported example. WDYT?

File: hacktoberfest_challenges/model_license_other.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+![](./assets/license_other_example.png)
+
+This challenge aims to improve the completeness of this metadata on the Hub, which will ultimately benefit all users.                                                                     
+
+## How to contribute?
+
+How to do it in practice? That's simple! We have listed models below that have `license:other` in their metadata and contain a LICENSE file in the repo. Those models require an update in their metadata to describe which license is attached.
+
+For each model, the workflow looks like this:
+1. Choose a model in the list below. For maximum impact, let's focus on the most downloaded or liked models first.
```
Comment (osanseviero) [inline hacktoberfest_challenges/model_license_other.md:30]:
```suggestion
1. Choose a model in the list below. We suggest focusing on the most downloaded or liked models first.
```

File: hacktoberfest_challenges/model_license_other.md [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+2. Check that the model doesn't already have a PR to update the license. Maybe another contributor already checked it!
```
Comment (osanseviero) [inline hacktoberfest_challenges/model_license_other.md:31]:
```suggestion
2. **Check that the model doesn't already have a PR to update the license**. Maybe another contributor already checked it!
```

**Review by osanseviero — APPROVED** [review]
Very nice!

File: hacktoberfest_challenges/model_license_other.md (hunk 2 above)
Comment (Wauplin) [reply hacktoberfest_challenges/model_license_other.md:8]: Oh, didn't know that. Do you have another example that is "quite known"?

Comment (osanseviero) [reply hacktoberfest_challenges/model_license_other.md:8]: Maybe coqui is a good example for that, but no strong opinion

Comment (osanseviero) [reply hacktoberfest_challenges/model_license_other.md:8]: Maybe coqui is a good example for that, but no strong opinion

**Review by osanseviero — APPROVED** [review]
Maybe also worth mentioning how to add an existing license. E.g. for many repos we can potentially switch to the llama2 license rather than other. Showing how to do that would be good as well

File: hacktoberfest_challenges/model_license_other.md (hunk 1 above)
Comment (Wauplin) [reply hacktoberfest_challenges/model_license_other.md:1]: Yes, right :+1:

**Comment by davanstrien** [comment]
I have uploaded task data/description for the dataset language ID task.

**Comment by Wauplin** [comment]
Thanks for the review @osanseviero. I updated the file to add special instructions for Llama 2 models (see https://github.com/huggingface/hub-docs/pull/981/commits/0c0edba48e773d6329f9a72721eb6210632c8fa9). Could you give it a quick look?

**Comment by Wauplin** [comment]
> I have uploaded task data/description for the dataset language ID task.

Oh indeed, that's why git didn't want me to push :smile: I think we did it at the same time. I'll have a look!

File: hacktoberfest_challenges/datasets_without_language.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## How to contribute?
+
+How can you help improve the coverage of language metadata on the Hub?
+
+For each dataset, the workflow is the following:
+
+1. Find a dataset that doesn't have the `language` field filled in. You can find a list of datasets without the `language` field filled in [here](#datasets-without-language-field-filled-in). We start with datasets that have the most downloads and likes.
+2. Check that the dataset doesn't already have a PR to add a language tag(s). Someone else may have already started working on it. You can check this by looking in the discussion section of the dataset page.
```
Comment (Wauplin) [inline hacktoberfest_challenges/datasets_without_language.md:30]:
```suggestion
2. **Check that the dataset doesn't already have a PR to add a language tag(s).** Someone else may have already started working on it. You can check this by looking in the discussion section of the dataset page. 
```

(for consistency with the other instructions)

File: hacktoberfest_challenges/datasets_without_language.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+   5. If the dataset viewer is available for the dataset, but you don't recognize the language, you can use the [facebook/fasttext-language-identification](https://huggingface.co/facebook/fasttext-language-identification) model or [Google Translate](https://translate.google.com/) to try to identify the language. 
+4. Once you've identified the language(s) of the dataset, you can add the language tag(s) to the dataset card. You can do this by clicking the `Edit` button on the dataset card. This will open a PR to the dataset repo. You can add the language tag(s) to the `language` field in the dataset card.
+5. Once done, open a PR on GitHub to update the table below. Once merged, this will count as a Hacktoberfest contribution! Add the `pr_url` (the one on the Hub) and a status (opened, merged, closed) in the PR. 
+6. Adding a language tag to some of the datasets below may not make sense. If so, add `not relevant` as the link in the `pr_url`. There may also be datasets where you need help with the language. In these cases, you can open a discussion to suggest a language tag(s) is added to the dataset. 
+
+## F.A.Q.
```
Comment (Wauplin) [inline hacktoberfest_challenges/datasets_without_language.md:41]: Another question I had: what about multiple-language datasets? is it worth adding a few languages even if I don't know if the list is exhaustive?

**Review by Wauplin — COMMENTED** [review]
"Datasets languages" challenge looks good to me. I would add a question in FAQ about multi-language datasets (as I was myself wondering about it when reading :smile:). Otherwise I think we're good to merge.

**Comment by davanstrien** [comment]
> "Datasets languages" challenge looks good to me. I would add a question in FAQ about multi-language datasets (as I was myself wondering about it when reading 😄). Otherwise I think we're good to merge.

Good idea, just added a clarification for this.

**Comment by Wauplin** [comment]
Merged!

**Pull request merged**